### PR TITLE
Fix | Fixed the issue with getColumns() only returning the first column against Azure DW

### PR DESCRIPTION
**Pull request by rene-ye**
Fix for #1192

### CODE REVIEW

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java [hunk 1]
```diff
@@ -756,6 +772,13 @@ private static String EscapeIDName(String inID) throws SQLServerException {
         }
     }
 
+    /*
+     * Helper method to create static maps used by the functions.
+     */
+    private static final SimpleEntry<Integer, String> createEntry(Integer i, String s) {
```
Comment (peterbae) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java]: where is this method being used?

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java [hunk 2]
```diff
@@ -605,6 +607,47 @@ private static String EscapeIDName(String inID) throws SQLServerException {
         return outID.toString();
     }
 
+    // Use LinkedHashMap to force retrieve elements in order they were inserted
+    private static final LinkedHashMap<Integer, String> getColumnsDWColumns = new LinkedHashMap<>();
+    static {
```
Comment (ulvii) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java]: I can see why you would want this block static, but I think a better option is to populate this map only if the connection is against Azure DW and it is empty.

**Comment by ulvii** [comment]
Needs a test.

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java [hunk 3]
```diff
@@ -675,6 +680,46 @@ private static String EscapeIDName(String inID) throws SQLServerException {
              * when user provides a different catalog than the one they're currently connected to. Will throw exception
              * when it's different and do nothing if it's the same/null.
              */
+            if (null == getColumnsDWColumns) {
```
Comment (peterbae) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java]: This should be in a synchronized block

Comment (ulvii) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java]: Please rename this map, there will be other APIs that require different column maps, so better the map name reflects the API name.

Comment (rene-ye) [reply src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java]: it's meant to reflect the method where it's used. This is for `getColumns`, if `getImportedKeys` needs one it'd be `getImportedKeysDWColumns`.

File: src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java [hunk 4]
```diff
@@ -608,6 +610,59 @@ public void testGetMaxConnections() throws SQLException {
...
+            try (ResultSet resultSet = databaseMetaData.getColumns(null, null, tableName, "%");) {
+                ResultSetMetaData rsmd = resultSet.getMetaData();
+                int rowCount = 0;
+                while (resultSet.next()) {
+                    for (int i = 1; i < 33; i++) {
+                        assertEquals(rsmd.getColumnName(i), getColumnsDWColumns.values().toArray()[i - 1]);
+                    }
+                    rowCount++;
+                }
+                assertEquals(rowCount, 3);
```
Comment (ulvii) [inline src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java]: assertEquals(3, rowCount);

**Pull request merged**